src/request/remove-key/RemoveKey.js

```diff
@@ -0,0 +1,150 @@
+        const $goToDownloadFile = ($removeKey.querySelector('#show-download-key-file'));
+        /** @type {HTMLButtonElement} */
+        const $goToShowRecoveryWords = ($removeKey.querySelector('#show-recovery-words'));
+        /** @type {HTMLButtonElement} */
+        const $goToRemoveKey = ($removeKey.querySelector('#remove-key-confirm'));
+        /** @type {HTMLFormElement} */
+        const $removeKeyPassphraseBox = ($removeKey.querySelector('.passphrase-box'));
+
+        // Components
+        this._removeKeyPassphraseBox = new PassphraseBox(
+            $removeKeyPassphraseBox, {
+                buttonI18nTag: 'passphrasebox-log-out',
+                bgColor: 'red',
+                hideInput: !this._request.keyInfo.encrypted,
+                minLength: this._request.keyInfo.hasPin ? 6 : undefined,
+            },
+        );
+
+        // events
+        $goToShowRecoveryWords.addEventListener('click', () => this._exportWordsHandler.run());
+        $goToDownloadFile.addEventListener('click', () => this._exportFileHandler.run());
+        $goToRemoveKey.addEventListener('click', () => {
+            $removeKey.classList.toggle('show-passphrase-box', true);
+            this._removeKeyPassphraseBox.reset();
+            this._removeKeyPassphraseBox.focus();
+        });
+        this._removeKeyPassphraseBox.on(PassphraseBox.Events.SUBMIT, this._passphraseSubmitted.bind(this));
+        this._removeKeyPassphraseBox.on(PassphraseBox.Events.CANCEL, () => {
+            $removeKey.classList.toggle('show-passphrase-box', false);
+        });
+
+        this._exportFileHandler.on(ExportFile.Events.EXPORT_FILE_KEY_CHANGED,
+            e => this._exportWordsHandler.setKey(e.key));
+        this._exportWordsHandler.on(ExportWords.Events.EXPORT_WORDS_KEY_CHANGED,
+            e => this._exportFileHandler.setKey(e.key, e.isProtected));
+    }
+
+    run() {
+        this._removeKeyPassphraseBox.reset();
+        window.location.hash = RemoveKey.Pages.REMOVE_KEY;
+        this._removeKeyPassphraseBox.focus();
+    }
+
+    /**
+     * @param {string} phrase
+     */
+    async _passphraseSubmitted(phrase) {
+        document.body.classList.add('loading');
+        try {
+            const passphraseBuffer = phrase ? Nimiq.BufferUtils.fromAscii(phrase) : undefined;
+            const key = await KeyStore.instance.get(this._request.keyInfo.id, passphraseBuffer);
+            if (!key) {
+                this._reject(new Error('keyId not found'));
+            }
+            await KeyStore.instance.remove(this._request.keyInfo.id);
+
+            /** @type {KeyguardRequest.SimpleResult} */
+            const result = {
+                success: true,
+            };
+            this._resolve(result);
+        } catch (e) {
+            if (e.message === 'Invalid key') this._removeKeyPassphraseBox.onPassphraseIncorrect();
+            else this._reject(e);
+        } finally {
+            document.body.classList.remove('loading');
```

Comment:
I think the loading state should only be removed in case of passphrase incorrect, as otherwise it re-enables the interface for the time it takes the browser to execute the RPC response.